src/modules/imageresizer/dll/Settings.cpp

```diff
@@ -1,66 +1,116 @@
 #include "pch.h"
-#include <commctrl.h>
 #include "Settings.h"
 
-const wchar_t c_rootRegPath[] = L"Software\\Microsoft\\ImageResizer";
-const wchar_t c_enabled[] = L"Enabled";
-const bool c_enabledDefault = true;
+#include <common/json.h>
+#include <common/settings_helpers.h>
+#include <filesystem>
+#include <commctrl.h>
 
-bool CSettings::GetEnabled()
+namespace
 {
-    return GetRegBoolValue(c_enabled, c_enabledDefault);
-}
+    const wchar_t c_imageResizerDataFilePath[] = L"\\image-resizer-settings.json";
+    const wchar_t c_rootRegPath[] = L"Software\\Microsoft\\ImageResizer";
+    const wchar_t c_enabled[] = L"Enabled";
 
-bool CSettings::SetEnabled(_In_ bool enabled)
-{
-    return SetRegBoolValue(c_enabled, enabled);
+    unsigned int GetRegNumber(const std::wstring& valueName, unsigned int defaultValue)
+    {
+        DWORD type = REG_DWORD;
+        DWORD data = 0;
+        DWORD size = sizeof(DWORD);
+        if (SHGetValue(HKEY_CURRENT_USER, c_rootRegPath, valueName.c_str(), &type, &data, &size) == ERROR_SUCCESS)
+        {
+            return data;
+        }
+        return defaultValue;
+    }
+
+    bool GetRegBoolean(const std::wstring& valueName, bool defaultValue)
+    {
+        DWORD value = GetRegNumber(valueName.c_str(), defaultValue ? 1 : 0);
+        return (value == 0) ? false : true;
+    }
+
+    bool LastModifiedTime(const std::wstring& filePath, FILETIME* lpFileTime)
+    {
+        WIN32_FILE_ATTRIBUTE_DATA attr{};
+        if (GetFileAttributesExW(filePath.c_str(), GetFileExInfoStandard, &attr))
+        {
+            *lpFileTime = attr.ftLastWriteTime;
+            return true;
+        }
+        return false;
+    }
 }
 
-bool CSettings::SetRegBoolValue(_In_ PCWSTR valueName, _In_ bool value)
+CSettings::CSettings()
 {
-    DWORD dwValue = value ? 1 : 0;
-    return SetRegDWORDValue(valueName, dwValue);
+    std::wstring result = PTSettingsHelper::get_module_save_folder_location(L"ImageResizer");
+    jsonFilePath = result + std::wstring(c_imageResizerDataFilePath);
+    Load();
 }
 
-bool CSettings::GetRegBoolValue(_In_ PCWSTR valueName, _In_ bool defaultValue)
+void CSettings::Save()
 {
-    DWORD value = GetRegDWORDValue(valueName, (defaultValue == 0) ? false : true);
-    return (value == 0) ? false : true;
+    json::JsonObject jsonData;
+
+    jsonData.SetNamedValue(c_enabled, json::value(settings.enabled));
+
+    json::to_file(jsonFilePath, jsonData);
+    GetSystemTimeAsFileTime(&lastLoadedTime);

```

Comment:
Should this be called here?

Reply:
Yes, because saving data to file is effectively changing his last modified time. If we don’t update this time here, next reload call will conclude that file changed since we loaded it last time and we need to reload again, even though we have up to date data. Maybe lastLoadedTime is not the best variable name, but nevertheless we need to update it both here and when parsing data.